### ISSUE TITLE
Misleading type error with implicit operator in generic type, from generic argument type to that generic type, when the generic argument is an interface

### ISSUE THREAD

**Issue by carlossanlop** [dotnet/roslyn#47399]
---
**Issue moved from dotnet/core#3939**
- Please respond to @markonius.

---

_From @markonius on Sunday, December 1, 2019 7:16:44 PM_

Fedora 31, dotnet 3.0.100

When using an implicit operator from a type variable `T` to a concrete type, if the type argument is an interface, a misleading error is shown: `Cannot implicitly convert type 'SomeType' to 'SomeGeneric<SomeType>'. An explicit conversion exists (are you missing a cast?)`
There is no explicit cast.

Here is a minimal example:

```cs
using System.Collections.Generic;
public class BadError<T> {
   public static implicit operator BadError<T>(T value) => new BadError<T>();
}
public class GoodError {
   // user-defined conversions to or from an interface are not allowed
   public static implicit operator GoodError(IReadOnlyList<int> value) => new GoodError();
}
public static class ObBoi {
   public static void Main()
   {
      IReadOnlyList<int> list = new List<int>();
      // Cannot implicitly convert type 'System.Collections.Generic.IReadOnlyList<int>' to 'BadError<System.Collections.Generic.IReadOnlyList<int>>'. An explicit conversion exists (are you missing a cast?)
      BadError<IReadOnlyList<int>> a = list;
   }
}
```

**Comment by CyrusNajmabadi** [comment]
I moved this to Roslyn.  Issues with error messages are under the purview of the compiler, not the language itself.  Thanks!
